File: frontend/src/Models/Location.ts
export interface PositionDetailsOSM {
	place_id: number;
	display_name: string;
	lat: string;
	lon: string;
	name: string;
}

export interface PositionDetails {
	place_id: number;
	display_name: string;
	lat: number;
	lon: number;
	name: string;
}

export interface PositionLatLon {
	lat: number;
	lon: number;
}

export interface PositionNameAdress {
	display_name: string;
}

export interface TaskLocation {
	id: number;
	place_id: number;
	display_name: string;
	name: string;
	lat: number;
	lon: number;
	geom: {};
	task_id: number;
}

export interface UserLocation {
	id: number;
	place_id: number;
	display_name: string;
	name: string;
	lat: number;
	lon: number;
	geom: {};
	user_id: number;
}

File: backend/backend/schemas/location.py
from pydantic import BaseModel, ConfigDict


class UserLocationSchema(BaseModel):
    place_id: int
    display_name: str
    name: str
    lat: float
    lon: float
    user_id: int


class TaskLocationSchema(BaseModel):
    place_id: int
    display_name: str
    name: str
    lat: float
    lon: float


class UserLocationPublic(BaseModel):
    id: int
    place_id: int
    display_name: str
    name: str
    lat: float
    lon: float
    geom: dict
    user_id: int
    model_config = ConfigDict(from_attributes=True)


class TaskLocationPublic(BaseModel):
    id: int
    place_id: int
    display_name: str
    name: str
    lat: float
    lon: float
    geom: dict
    task_id: int
    model_config = ConfigDict(from_attributes=True)


class LocationList(BaseModel):
    locations: list[TaskLocationPublic]
